go/vt/vitessdriver/driver.go

```diff
@@ -225,25 +182,29 @@ func (c *conn) dial() error {
 	return err
 }
 
-func (c *conn) useExecuteShards() bool {
-	return c.Keyspace != "" && c.Shard != ""
-}
-
 func (c *conn) Prepare(query string) (driver.Stmt, error) {
 	return &stmt{c: c, query: query}, nil
 }
 
+func (c *conn) PrepareContext(ctx context.Context, query string) (driver.Stmt, error) {
+	return c.Prepare(query)
+}
+
 func (c *conn) Close() error {
 	c.vtgateConn.Close()
 	return nil
 }
 
 func (c *conn) Begin() (driver.Tx, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+	return c.BeginContext(ctx)
+}
+
+func (c *conn) BeginContext(ctx context.Context) (driver.Tx, error) {
 	if c.Streaming {
 		return nil, errors.New("transaction not allowed for streaming connection")
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
-	defer cancel()
 	tx, err := c.vtgateConn.Begin(ctx)
```

Comment:
The Vitess isolation model is different from the one exposed by a traditional database. We have non-transactional reads that can be consistent or eventually consistent. Then we have traditional transactions with serializable writes. The transactional reads depend on whether the query is single-shard or not, and whether you execute them as 'FOR UPDATE'. In simpler terms, these are along the lines of what a key-value store supports.

Long story short, we don't support any of the isolation levels as defined by their contracts. So, I thought it's better to just ignore that aspect. The other option is to explicitly ensure that no isolation level was requested.